shell/components/Resource/Detail/ResourceRow.vue

```diff
@@ -0,0 +1,119 @@
+<script lang="ts">
+import SubtleLink from '@shell/components/SubtleLink.vue';
+import { StateColor, stateColorCssVar } from '@shell/utils/style';
+import { sortBy } from 'lodash';
+import { RouteLocationRaw } from 'vue-router';
+
+export interface Count {
+  label: string;
+  count: number;
+}
+
+export interface Props {
+  label: string;
+  to?: RouteLocationRaw;
+  color?: StateColor;
+  counts?: Count[];
+}
+
+export function extractCounts(labels: string[]): Count[] {
+  const accumulator: { [k: string]: number} = {};
+
+  labels.forEach((l: string) => {
+    accumulator[l] = accumulator[l] || 0;
+    accumulator[l]++;
+  });
+
+  const counts: Count[] = Object.entries(accumulator).map(([label, count]) => ({ label, count }));
+
+  return sortBy(counts, 'label');
+}
+
+</script>
+
+<script setup lang="ts">
+const {
+  label, to, counts, color
+} = defineProps<Props>();
+</script>
+
+<template>
+  <div class="resource-row">
+    <div class="left">
+      <SubtleLink
+        v-if="to"
+        :to="to"
+      >
+        {{ label }}
+      </SubtleLink>
+      <span
+        v-else
+        class="text-muted"
+      >
+        {{ label }}
+      </span>
+    </div>
+    <div class="right">
+      <div
+        v-if="!counts || counts.length == 0"
+        class="text-muted"
+      >
+        &mdash;
+      </div>
+      <div
+        v-else
+        class="counts"
+      >
+        <span
+          v-if="color"
+          class="dot"
+          :style="{backgroundColor: stateColorCssVar(color)}"
+        >
+&nbsp;
+        </span>
+        <span
+          v-for="(count, i) in counts"
+          :key="i"
```

Comment:
We generally want to avoid using the index as a key in `v-for`, but I think this looks safe; the content appears to be static.

Reply:
We can just go with the count.label for the key. I'll switch it up.